### PR TITLE
Weekly patch release: 1.7.6

### PR DESCRIPTION
**Pull request by krshrimali**
## What does this PR do?

Preparing for the weekly patch release, 1.7.6. Should go out on Tuesday (13th September 2022).

```bash
gh pr list -s merged -S 'merged:2022-09-06T22:00:00.000Z..2022-09-13T22:00:00.000Z' --json mergedAt,milestone,url,mergeCommit,title --jq 'sort_by(.mergedAt) | reverse | .[] | select((.milestone.title == "pl:1.7.x") or (.milestone.title == null) or (.milestone.title == "all:next")) | [.url, .mergeCommit.oid, .title] | join(" ")' --limit 100

https://github.com/Lightning-AI/lightning/pull/14637 19a1274093f0a406d2a190c1947eab892ccdc106 Better error message when dataloader and datamodule is None (V2)
https://github.com/Lightning-AI/lightning/pull/14599 1ee3d1eb72f61b05353e289727cf6e20791b4a64 Avoid warning when cloning tensor in self.log
https://github.com/Lightning-AI/lightning/pull/14597 1680a768197fee78d30dd26dd3feef83c10db6bc Removed from_argparse_args tests in test_cli.py
https://github.com/Lightning-AI/lightning/pull/14665 3d540efe4fa16c8adede16cb0761eda3b3acc58f Update psutil requirement from <=5.9.1 to <5.9.3 in /requirements
https://github.com/Lightning-AI/lightning/pull/14565 92309c215339da9c1913b782dccf7b73a5c14532 Remove skipping logic in PL CI
https://github.com/Lightning-AI/lightning/pull/14625 dfa570ef9fa8c292b4d426e24b053618d521d1c2 Run CircleCI with the HEAD sha, not the base
https://github.com/Lightning-AI/lightning/pull/14657 cf3428784f4f51862eb2c098a7ac1244397ebb24 Set `running_torchscript` recursively
https://github.com/Lightning-AI/lightning/pull/14617 a409b2305511b655352593c74cf064322f3a1090 Remove deprecated test_tube dependency from `environment.yml`
https://github.com/Lightning-AI/lightning/pull/14529 40868f7f43f308ca3b35b1e46a78bfa46fb940d7 Add bagua support for CUDA 11.6 images
https://github.com/Lightning-AI/lightning/pull/14627 7cbf153332707b1677ac37d300c68659aa2a90c0 CI: hotfix last version
https://github.com/Lightning-AI/lightning/pull/14576 4d082e4dac7fec388bdc49578f1d50ae72e498fb Update fastapi requirement from <=0.79.0 to <0.83.0 in /requirements
https://github.com/Lightning-AI/lightning/pull/14582 6cf666c6f16f5ffe550efbca79e3e7f96a086a2d Update neptune-client requirement from <0.16.4,>=0.10.0 to >=0.10.0,<0.16.8 in /requirements
https://github.com/Lightning-AI/lightning/pull/14454 31dc6c67142ad1c1e4fa192cad320a99114dc4c1 Add compatibility when `torch.distributed` is not available
https://github.com/Lightning-AI/lightning/pull/14543 46519e2fc770f3393c7f5e72e6f36a01a15efb27 Add path filters to the TPU job
https://github.com/Lightning-AI/lightning/pull/14505 6de2b0b5284eb3c27583e4d8dc9017675a023b28 Docs [Fix]: use bytes instead of strings while writing
https://github.com/Lightning-AI/lightning/pull/14288 dbd7703e88c39421d740c010cac7f0bb23981c12 Update fsspec[http] requirement from !=2021.06.0,<2022.6.0,>=2021.05.0 to >=2021.05.0,!=2021.06.0,<2022.8.0 in /requirements
```

PR carried over, wasn't released in 1.7.4 and forgotten in 1.7.5: https://github.com/Lightning-AI/lightning/pull/14372

```bash
gh pr list -s merged -S 'merged:2022-06-09T22:00:00.000Z..2022-09-13T22:00:00.000Z' --json author,milestone --jq '.[] | select((.milestone.title == "pl:1.7.x") or (.milestone.title == null) or (.milestone.title == "all:next")) | ["@" + .author.login] | join("")' --limit 100 | sort --ignore-case | uniq | xargs echo

@akihironitta @awaelchli @Borda @carmocca @dependabot @krshrimali @mauvilsa @pierocor @rohitgr7 @wangraying

```

### Does your PR introduce any breaking changes? If yes, please list them.

Nopes!

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

**Comment by awaelchli** [comment]
I'm looking to get these in:

https://github.com/Lightning-AI/lightning/pull/14599
https://github.com/Lightning-AI/lightning/pull/14637

File: src/pytorch_lightning/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,22 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
+## [1.7.6] - 2022-09-13
+
+### Changed
+
+- When using multiple loggers, by default checkpoints and profiler output now get saved to the log dir of the first logger in the list ([#14325](https://github.com/Lightning-AI/lightning/pull/14325))
```
Comment (awaelchli) [inline src/pytorch_lightning/CHANGELOG.md:11]: #14325 This has milestone 1.8, it should not be included.

**Review by awaelchli — APPROVED** [review]

File: src/pytorch_lightning/CHANGELOG.md (hunk 1 above)
Comment (krshrimali) [reply src/pytorch_lightning/CHANGELOG.md:11]: Sorry, I'll remove this! :))

Comment (krshrimali) [reply src/pytorch_lightning/CHANGELOG.md]: Done! @awaelchli - I hope it's okay if I remove and amend the commit to the latest commit; https://github.com/Lightning-AI/lightning/pull/14618/commits/2f6bcbc44eb3859a9fa051c7486fde1838711d7d. Please let me know if it's not okay. :)

File: tests/tests_pytorch/test_cli.py [hunk 2]
```diff
@@ -67,42 +63,6 @@ def mock_subclasses(baseclass, *subclasses):
         yield None
 
 
-@mock.patch("argparse.ArgumentParser.parse_args")
-def test_default_args(mock_argparse):
```
Comment (carmocca) [inline tests/tests_pytorch/test_cli.py:71]: Why are these CLI tests removed? Bad merge?

Comment (krshrimali) [reply tests/tests_pytorch/test_cli.py:71]: @carmocca - Thanks for cross-checking, this comes from: https://github.com/Lightning-AI/lightning/pull/14597/. So probably not a bad merge.

Comment (carmocca) [reply tests/tests_pytorch/test_cli.py:71]: My bad. I missed that PR

**Review by carmocca — APPROVED** [review]

**Pull request merged**